daemons/controld/controld_fencing.c

```diff
@@ -847,11 +848,11 @@ te_fence_node(crm_graph_t *graph, crm_action_t *action)
                                   (int) (transition_graph->stonith_timeout / 1000),
                                   0);
 
+    transition_key = pcmk__transition_key(transition_graph->id, action->id, 0,
+                                          te_uuid),
     stonith_api->cmds->register_callback(stonith_api, rc,
                                          (int) (transition_graph->stonith_timeout / 1000),
-                                         st_opt_timeout_updates,
-                                         generate_transition_key(transition_graph->id, action->id,
-                                                                 0, te_uuid),
```

Comment:
I haven't dug into it further, but do the results of pcmk__transition_key need to be freed?  The fact that it's calling crm_strdup_printf would make me think yes, but I don't know what the callback is going to do with the data.  There are several other places in this patch where the same question would apply.

Reply:
Good eye. In these cases that's the user data passed to the callback, and the callback frees it. I'm not thrilled that it's non-obvious but it's correct.